src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -174,6 +179,90 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
                 boundNamedArguments, resultKind, attributeType, hasErrors: resultKind != LookupResultKind.Viable);
         }
 
+        private void CheckArgumentNullability(MethodSymbol attributeConstructor, AnalyzedAttributeArguments arguments, DiagnosticBag diagnostics)
+        {
+            // Check constructor arguments
+            for (int i = 0; i < arguments.ConstructorArguments.Arguments.Count; i++)
+            {
+                var argument = arguments.ConstructorArguments.Argument(i);
+                var argName = arguments.ConstructorArguments.Name(i);
+
+                // map the argument back to the correct parameter
+                var parameter = argName == null
+                                ? attributeConstructor.Parameters[i]
+                                : attributeConstructor.Parameters.Single(p => p.Name == argName);
+
+                reportPossibleNullabilityMismatch(parameter.Type, argument, parameter.IsParams);
+            }
+
+            // Check assignments
+            foreach (BoundAssignmentOperator assignment in arguments.NamedArguments)
+            {
+                TypeSymbolWithAnnotations assignmentType;
+                switch (assignment.Left)
+                {
+                    case BoundPropertyAccess propertyAccess:
+                        assignmentType = propertyAccess.PropertySymbol.Type;
+                        break;
+                    case BoundFieldAccess fieldAccess:
+                        assignmentType = fieldAccess.FieldSymbol.Type;
+                        break;
+                    default:
+                        continue; 
+                }
+
+                reportPossibleNullabilityMismatch(assignmentType, assignment.Right);
+            }
+
+            void reportPossibleNullabilityMismatch(TypeSymbolWithAnnotations destination, BoundExpression sourceExpression, bool destinationIsParams = false)
+            {
+                if (IsNullableEnabled(sourceExpression.SyntaxTree, sourceExpression.Syntax.SpanStart))
+                {
+                    if (destination.NullableAnnotation == NullableAnnotation.NotAnnotated &&
+                       sourceExpression.ConstantValue == ConstantValue.Null &&
+                       !isSupressed(sourceExpression) &&
+                       !destinationIsParams)
```

Comment:
>destinationIsParams [](start = 24, length = 19)

Do we need to know whether the constructor was invoked in normal or expanded form? For instance, with `A(params object?[])`, the cases `A(null)` and `A((object[])null)`.